src/views/tutorial-view/components/mdx-img/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { ReactElement } from 'react'
+import { MdxImgProps } from './types'
+import classNames from 'classnames'
+import s from './mdx-img.module.css'
+
+function MdxImg({
+  src,
+  alt,
+  title,
+  noMargin,
```

Comment:
`noMargin` and `noBorder` are here for use with `ImageConfig`
- `noMargin` - when used with a caption, we want the `img` element to not have any margin (so that we can set margin once on the enclosing `figure`)
- `noBorder` - we want to deprecate this, but it's something in `learn` already so figured we need to support it until we can make the necessary content lift

Reply:
👆 I think solving the `noMargin` case with CSS would remove our model of having component styles encapsulated within each component. When we start writing styles that target HTML primitives in CSS modules, we complicate the guarantee in any nested components that their styles will not be affected by parent components.

In this specific case this is relatively trivial - at present, we don't expect anything but `<img />` to be rendered within `figure`. Even so, there could be a relevant scenario on the horizon - it's likely we'll eventually support more arbitrary `caption` content rather than just plain text (for example, we could support links in captions). What if we have a case where we want to show an inline "external link" icon in the `caption` area? Then our `.figure img` CSS selector inadvertently targets that element.

I don't mean to suggest that this scenario is necessarily likely, or that we'll be able to come up with such examples in every possible case. But this aim for full encapsulation was the broader reasoning behind our adoption of CSS modules. There's a bit more detail on this specific case in Notion as well (search for `Issue: Generic selectors can be hard to override`).

Let me know if that context is helpful, happy to add anything else I can, or revisit the approach here if that rationale seems off 🙇‍♂️

Reply:
Sounds good! No firm opinion on this specific thing, just looking for context. I do have opinions on what we render for `caption`, so I'm glad we're forcing that to be a string for now, but that's for a later discussion. 😊